influxdb3_write/src/write_buffer/buffer_segment.rs

```diff
@@ -507,7 +727,8 @@ impl ClosedBufferSegment {
                             db_name,
                             &table.name,
                             &table_buffer.segment_key.to_string(),
-                            self.segment_id.0,
+                            self.segment_id,
+                            table_parquet_files.parquet_files.len() as u32 + 1,
```

Comment:
I don't think you need the +1 here for the file name. The length should suffice.

Reply:
It would, I just had a mental block against file numbers starting at 0 since this is before the file gets added to the collection.